### PR TITLE
Make withExceptionHandling useful by returning the result

### PR DESCRIPTION
**Pull request by elpete**

### CODE REVIEW

**Comment by jclausen** [comment]
@elpete On adobe engines, you're going to need to use `cfconfig` to set the `errorStatusCode` to `true` in order for your tests to pass.  Out of the box, ACF returns a 200 on errors - don't ask me why.

**Comment by bdw429s** [comment]
> Out of the box, ACF returns a 200 on errors - don't ask me why.

it's a security "feature" meant to prevent hackers from finding out via a script if they've caused an error on your site.  That said, I hate it because it's dumb and HTTP status codes should always represent what happened in the request.  I have seen security scanners give "false positives" just because they were able to get a 500 status code back.

File: system/testing/BaseTestCase.cfc [hunk 1]
```diff
@@ -835,6 +843,32 @@ component extends="testbox.system.compat.framework.TestCase"  accessors="true"{
         } else {
             return getPageContext().getResponse().getResponse();
         }
+    }
+
+    /**
```
Comment (lmajano) [inline system/testing/BaseTestCase.cfc]: @elpete can you remove this, since we talked about leveraging the exception bean approach.

Comment (lmajano) [reply system/testing/BaseTestCase.cfc:848]: @elpete I want to merge this, but can you provide some cleanup please.

**Review by lmajano — CHANGES_REQUESTED** [review]

**Comment by elpete** [comment]
I don't remember what I was trying to do here.  We're probably better starting from scratch. :sweat_smile:

**Pull request closed without merging**